### DISCUSSION TITLE
changeData更新数据tooltip会消失

### DISCUSSION THREAD

**Discussion by demolhy** [antvis/G2#5884] [Q&A]
### 问题描述

需要重新移动鼠标才会显示，而且偶尔会跑到左下角去再回来
````
chart
.line()
.axis('x', {
  label: null,
  title: null,
  tickFilter: false,
  tick: false,
  line: false,
  grid: true,
  gridFilter: (_: any, index: number) => {
    return index % 10 === 0;
  },
})
.axis('y', {
  labelFormatter: (val: string) => {
    return `${val}%`;
  },
  line: true,
  lineExtension: [0, 0],
  lineStroke: '#EEEEEE',
})
.scale('y', { zero: true, type: 'linear', domain: [0, 100] })
.encode('x', 'time')
.encode('y', 'value')
.encode('shape', 'smooth')
.animate(false)
.encode('color', 'group')
.scale('color', {
  range: ['#00baa4', '#0090f8', '#f97316'],
  independent: true,
})
.interaction('tooltip', {
  render: (_: unknown, { title, items }: never) => tooltipRender(title, items),
})
.tooltip(
  (
    d, // 每一个数据项
  ) => {
    return {
      value: d.value + '%',
    };
  },
)
.style('strokeWidth', 1)
````

### 预期行为

实时显示更新的tooltip数据，tooltip不消失

### 平台

- 操作系统: [macOS, Windows, Linux, React Native ...]
- 网页浏览器: [Google Chrome, Safari, Firefox]

### 屏幕截图或视频（可选）

​

https://github.com/antvis/G2/assets/39995045/bda5caaf-671e-46fb-a2c2-5868645c3304

**Comment by pearmini** [comment]
这是符合预期的，因为每次渲染都会销毁 tooltip。如果希望模拟不消失的效果，可以再每次调用 chart.changeData 之后，调用 `chart.emit('tooltip', {})`，具体参考[文档](https://g2.antv.antgroup.com/spec/interaction/tooltip#%E6%89%8B%E5%8A%A8%E6%8E%A7%E5%88%B6%E5%B1%95%E7%A4%BA%E9%9A%90%E8%97%8F)。

**Reply by demolhy** [reply]
emit的参数文档有更详细的吗,或者有无案例参考
````
chart.emit('tooltip:show', {
  data: {
    // 会找从原始数据里面找到匹配的数据
    data: { x: 'xValue' },
  },
});
````
没有效果

**Reply by pearmini** [reply]
这里有一个可以参考的案例：https://g2.antv.antgroup.com/examples/component/tooltip/#tooltip-click
